test/release/examples/primers/modules.chpl

```diff
@@ -0,0 +1,390 @@
+/*
+ * Module Primer
+ *
+ * This primer introduces the concept of modules, a strategy for encapsulating
+ * code for later use.  It covers:
+ *   how to define a module
+ *   namespace control for a module's symbols
+ *   outside access of a module's symbols
+ *   namespace control when using a module in an outside context, including:
+ *     unlimited
+ *     explicit exclusion of symbols
+ *     explicit inclusion of symbols
+ *     renaming included symbols
+ *   outside access of modules in different files
+ */
+
+
+/* A module is a grouping of code - each program consists of at least one
+   module, and every symbol is associated with some module.  If all the code of
+   a file is not enclosed in an explicit module, defined using the syntax below,
+   then the file itself is treated as a module with the same name as the file
+   (minus the .chpl suffix).
+ */
+module modToUse {
+
+  /* In this case, foo is a public global variable which lives in the module
+     modToUse */
+  var foo = 12;
+
+  // As is bar.
+  var bar: int = 2;
+
+  /* baz is a public function which lives in modToUse */
+  proc baz (x, y) {
+    return x * (x + y);
+  }
+
+  /* A module defined within another module is known as a nested module.  These
+     modules can take advantage of symbols defined within their parent module,
+     but their parent module must take additional steps to access the contents
+     of the nested module.  We will cover how to access the contents of other
+     modules later in this primer.
+  */
```

Comment:
"known as" -> "called"?
"take advantage of" -> "refer to"
"take additional steps to access" -> "can't directly access"
Maybe find a way to introduce the term "sub-module" here?  e.g., "These submodules can..."

Order-wise, it seems like in a "read primers from top to bottom" world, it might be nice to cover simple cases of module 'use's before getting into nested modules.  In particular, it would avoid the vague forward reference to "additional steps" and make it seem more like the way anybody accesses any module that they can see.  But that rearrangement may be more work than it's worth.